src/main/java/com/Novi/TechItEasy/services/TelevisionService.java

```diff
@@ -145,6 +151,85 @@ public TelevisionDto changeTelevision(Long id, TelevisionInputDto television) {
     }
 
 
+    ///// For assigning remote controllers to television by id /////
+    public TelevisionDto assignRemoteControllerToTelevision(Long televisionId, IdInputDto remoteId) {
+        Optional<Television> fetchedTelevision =  televisionRepository.findById(televisionId);
+        Optional<RemoteController> fetchedRemoteController = remoteControllerRepository.findById(remoteId.id);
+
+        if (fetchedTelevision.isPresent()) {
+            if (fetchedRemoteController.isPresent()) {
+
+                fetchedTelevision.get().setRemoteController(fetchedRemoteController.get());
+
+                televisionRepository.save(fetchedTelevision.get());
+                fetchedRemoteController.get().setTelevision(fetchedTelevision.get());
+
+                return TelevisionDto.fromTelevision(televisionRepository.findById(televisionId).get());
+            } else {
+                throw new RecordNotFoundException("We hebben geen afstandsbediening met ID: " + remoteId.id + ".");
+            }
+        } else {
+            throw new RecordNotFoundException("We hebben geen televisie met ID: " + televisionId + ".");
+        }
+    }
+
+
+    ///// For assigning CI module to television by id /////
+    public TelevisionDto assignCiModuleToTelevision(Long televisionId, IdInputDto ciModuleId) {
+        Optional<Television> fetchedTelevision =  televisionRepository.findById(televisionId);
+        Optional<CI_Module> fetchedCiModule = ci_moduleRepository.findById(ciModuleId.id);
+
+        if (fetchedTelevision.isPresent()) {
+            if (fetchedCiModule.isPresent()) {
+                List<Television> ciModuleTelevisionList = fetchedCiModule.get().getTelevisionList();
+
+                fetchedTelevision.get().setCi_module(fetchedCiModule.get());
+                televisionRepository.save(fetchedTelevision.get());
+
+                ciModuleTelevisionList.add(fetchedTelevision.get());
+
+                fetchedCiModule.get().setTelevisionList(ciModuleTelevisionList);
+                ci_moduleRepository.save(fetchedCiModule.get());
```

Comment:
Ook hier is het niet nodig om de relatie in ci_module op te slaan, omdat television de owner van de releatie is. Voor WallBracket geldt hetzelfde